File: Projection/demo/src/main/java/com/kuanquan/demo/DLNAService.java
package com.kuanquan.demo;

import android.app.Service;
import android.content.*;
import android.net.wifi.WifiManager;
import android.os.IBinder;
import android.util.Log;

import com.plutinosoft.platinum.ServerParams;

/**
 * The service that manage the server instance
 * 管理服务器实例的服务
 * TODO 2
 */
public class DLNAService extends Service {

    public static final String EXTRA_SERVER_PARAMS = "EXTRA_SERVER_PARAMS";

    private static final String TAG = "DLNAService";
    private WifiManager.MulticastLock mMulticastLock;

    @Override
    public IBinder onBind(Intent intent) {
        return null;
    }

    @Override
    public void onCreate() {
        super.onCreate();
        acquireMulticastLock();
        buildNotification();
    }

    private void buildNotification() {
//        Intent intent = new Intent(this, MainActivity.class);
//        intent.setFlags(Intent.FLAG_ACTIVITY_REORDER_TO_FRONT);
//        mNotification = NotificationHelper.INSTANCE
//                .getNotification(intent, getString(R.string.server_notification_title),
//                        getString(R.string.server_notification_text));
    }

    private void acquireMulticastLock() {
        WifiManager wifiManager = (WifiManager) getApplicationContext().getSystemService(Context.WIFI_SERVICE);
        if (wifiManager != null) {
            mMulticastLock = wifiManager.createMulticastLock(TAG);
            mMulticastLock.acquire();
        }
    }

    @Override
    public int onStartCommand(Intent intent, int flags, int startId) {
        if (intent != null) {
            ServerParams params = intent.getParcelableExtra(EXTRA_SERVER_PARAMS);
            if (params != null) {
                Log.e("DLNAService", "开启服务");
                ServerInstance.INSTANCE.start(params);
//                NotificationHelper.INSTANCE.notify(mNotification);
            }
        }
        return super.onStartCommand(intent, flags, startId);
    }

    @Override
    public void onDestroy() {
        if (mMulticastLock != null) {
            mMulticastLock.release();
            mMulticastLock = null;
        }
        ServerInstance.INSTANCE.stop();
//        NotificationHelper.INSTANCE.cancel();
        super.onDestroy();
    }
}

File: Projection/test/src/main/java/com/kuanquan/test/DLNAService.kt
package com.kuanquan.test

import android.app.Service
import android.content.*
import android.net.wifi.WifiManager
import android.net.wifi.WifiManager.MulticastLock
import android.os.IBinder
import android.util.Log
import com.plutinosoft.platinum.ServerParams

/**
 * The service that manage the server instance
 * 管理服务器实例的服务
 * TODO 2
 */
class DLNAService: Service() {

    private val TAG = "DLNAService"

    companion object{
        // 从 MainActivity 传递到 DLNAService 的参数的 key
        val EXTRA_SERVER_PARAMS_KEY = "EXTRA_SERVER_PARAMS_KEY"
    }

    private var mMulticastLock: MulticastLock? = null

    override fun onBind(intent: Intent?): IBinder? {
        return null
    }

    override fun onCreate() {
        super.onCreate()
        acquireMulticastLock()
    }

    private fun acquireMulticastLock() {
        val wifiManager = applicationContext.getSystemService(Context.WIFI_SERVICE) as? WifiManager
        mMulticastLock = wifiManager?.createMulticastLock(TAG)
        mMulticastLock?.acquire()
    }

    override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
            val params: ServerParams? = intent?.getParcelableExtra(EXTRA_SERVER_PARAMS_KEY)
            if (params != null) {
                Log.e("DLNAService", "开启服务")
                ServerInstance.start(params)
            }
        return super.onStartCommand(intent, flags, startId)
    }

    override fun onDestroy() {
        mMulticastLock?.release()
        mMulticastLock = null
        ServerInstance.stop()
        super.onDestroy()
    }
}